### PR TITLE
8316415: Parallelize sun/security/rsa/SignedObjectChain.java subtests

### PR DESCRIPTION
**Pull request by msobiers**
sun/security/rsa/SignedObjectChain.java is very slow when run with C1, I suspect because some crypto intrinsics are only implemented in C2. Commit contains changes made to parallelize it.

Comparison of before and after parallelization:
time make test TEST=jdk/sun/security/rsa/SignedObjectChain.java TEST_VM_OPTS="-XX:+UseParallelGC -XX:TieredStopAtLevel=1"
before:   270.72s user 4.88s system 108% cpu 4:14.43 total
after:   410.76s user 7.50s system 555% cpu 1:15.23 total
after second commit:   375.46s user 4.59s system 539% cpu 1:10.41 total

time make test TEST=jdk/sun/security/rsa/SignedObjectChain.java TEST_VM_OPTS="-XX:+UseParallelGC"
before:   63.67s user 4.67s system 161% cpu 42.424 total
after:   130.36s user 7.47s system 585% cpu 23.526 total
after second commit:   67.31s user 4.48s system 417% cpu 17.183 total

time make test TEST=jdk/sun/security/rsa/SignedObjectChain.java TEST_VM_OPTS="-XX:+UseShenandoahGC -XX:TieredStopAtLevel=1"
before:   281.99s user 5.54s system 108% cpu 4:24.09 total
after:   386.98s user 8.62s system 496% cpu 1:19.73 total
after second commit:   413.51s user 5.08s system 613% cpu 1:08.25 total

time make test TEST=jdk/sun/security/rsa/SignedObjectChain.java TEST_VM_OPTS="-XX:+UseShenandoahGC"
before:   65.86s user 5.05s system 156% cpu 45.215 total
after:   135.90s user 7.66s system 585% cpu 24.502 total
after second commit:   83.25s user 4.82s system 469% cpu 18.741 total

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8316415](https://bugs.openjdk.org/browse/JDK-8316415): Parallelize sun/security/rsa/SignedObjectChain.java subtests (**Enhancement** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**) ⚠️ Review applies to [0040ba31](https://git.openjdk.org/jdk/pull/15860/files/0040ba31e681fbf5e59a0655a7bd7721e101c4f4)
 * [Valerie Peng](https://openjdk.org/census#valeriep) (@valeriepeng - **Reviewer**) ⚠️ Review applies to [0040ba31](https://git.openjdk.org/jdk/pull/15860/files/0040ba31e681fbf5e59a0655a7bd7721e101c4f4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15860/head:pull/15860` \
`$ git checkout pull/15860`

Update a local copy of the PR: \
`$ git checkout pull/15860` \
`$ git pull https://git.openjdk.org/jdk.git pull/15860/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15860`

View PR using the GUI difftool: \
`$ git pr show -t 15860`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15860.diff">https://git.openjdk.org/jdk/pull/15860.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15860#issuecomment-1729507337)

### CODE REVIEW

File: test/jdk/sun/security/rsa/SignedObjectChain.java [hunk 1]
```diff
@@ -52,13 +53,13 @@ public Test(Chain.SigAlg sigAlg) {
     };
 
     public static void main(String argv[]) {
-        boolean resutl = java.util.Arrays.stream(tests).allMatch(
-                (test) -> Chain.runTest(test));
+        boolean result = Arrays.stream(tests).parallel().allMatch(Chain::runTest);
 
-        if(resutl) {
+        if(result) {
```
Comment (shipilev) [inline test/jdk/sun/security/rsa/SignedObjectChain.java:58]:
```suggestion
        if (result) {
```

File: test/jdk/sun/security/rsa/SignedObjectChain.java [hunk 2]
```diff
@@ -52,13 +53,13 @@ public Test(Chain.SigAlg sigAlg) {
...
             System.out.println("All tests passed");
         } else {
             throw new RuntimeException("Some tests failed");
         }
     }
 }
+
```
Comment (shipilev) [inline test/jdk/sun/security/rsa/SignedObjectChain.java:65]: Do we need a newline here?

**Review by shipilev — APPROVED** [review]
Looks nice. A few stylistic comments:

**Comment by shipilev** [comment]
Shows a very good improvement here as well:

```
% time CONF=linux-x86_64-server-fastdebug make test TEST=sun/security/rsa/SignedObjectChain.java TEST_VM_OPTS="-XX:TieredStopAtLevel=1"

Baseline:
CONF=linux-x86_64-server-fastdebug make test    333.34s user 5.04s system 111% cpu 5:02.33 total
CONF=linux-x86_64-server-fastdebug make test    292.03s user 5.27s system 114% cpu 4:20.71 total
CONF=linux-x86_64-server-fastdebug make test    330.51s user 5.23s system 111% cpu 5:00.18 total

Patched:
CONF=linux-x86_64-server-fastdebug make test    309.72s user 4.92s system 505% cpu 1:02.25 total
CONF=linux-x86_64-server-fastdebug make test    275.63s user 5.02s system 540% cpu   51.82 total
CONF=linux-x86_64-server-fastdebug make test    299.18s user 5.06s system 546% cpu   55.64 total
```

**Comment by shipilev** [comment]
@valeriepeng, would you like to take a look?

**Comment by shipilev** [comment]
Or maybe @rhalade?

**Review by rhalade — APPROVED** [review]

**Comment by valeriepeng** [comment]
I will take a look, sorry has been busy the past few days...

File: test/jdk/sun/security/rsa/SignedObjectChain.java (hunk 1 above)
Comment (valeriepeng) [reply test/jdk/sun/security/rsa/SignedObjectChain.java:58]: +1

File: test/jdk/sun/security/rsa/SignedObjectChain.java (hunk 2 above)
Comment (valeriepeng) [reply test/jdk/sun/security/rsa/SignedObjectChain.java:65]: +1, would be nice if this extra newline is removed.

**Comment by valeriepeng** [comment]
Looks good to me as well. Would be nice to apply the stylistic changes suggested by @shipilev  before integrating. Thanks!

**Review by valeriepeng — APPROVED** [review]

**Review by msobiers — COMMENTED** [review]
Removing unnecessary line

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**